mkosi/distributions/gentoo.py

```diff
@@ -211,43 +109,39 @@ def fetch_fix_stage3(self) -> None:
             else:
                 die("profile names changed upstream?")
 
+        stage3_tar_path = state.cache_dir / stage3_tar
+        cls.stage3_cache = stage3_tar_path.with_name(stage3_tar.name).with_suffix(".tmp")
+
+        user_config_path = cls.stage3_cache / USER_CONFIG_PATH
+
+        for d in ("binpkgs", "distfiles", "repos"):
+            state.cache_dir.joinpath(d).mkdir(exist_ok=True)
+
         stage3_url_path = urllib.parse.urljoin(
-            self.portage_cfg["GENTOO_MIRRORS"],
-            f"releases/{self.arch}/autobuilds/{stage3_tar}",
+            state.config.mirror, f"releases/{arch}/autobuilds/{stage3_tar}",
         )
-        stage3_tar_path = self.state.cache_dir / stage3_tar
-        stage3_tmp_extract = stage3_tar_path.with_name(stage3_tar.name + ".tmp")
         if not stage3_tar_path.is_file():
             log_step(f"Fetching {stage3_url_path}")
             stage3_tar_path.parent.mkdir(parents=True, exist_ok=True)
             urllib.request.urlretrieve(stage3_url_path, stage3_tar_path)
 
-        stage3_tmp_extract.mkdir(parents=True, exist_ok=True)
-
-        with flock(stage3_tmp_extract):
-            if not stage3_tmp_extract.joinpath(".cache_isclean").exists():
-                log_step(f"Extracting {stage3_tar.name} to {stage3_tmp_extract}")
-
-                run([
-                    "tar",
-                    "--numeric-owner",
-                    "-C", stage3_tmp_extract,
-                    "--extract",
-                    "--file", stage3_tar_path,
-                    "--exclude", "./dev",
-                ])
+        cls.stage3_cache.mkdir(parents=True, exist_ok=True)
 
-                unlink_try_hard(stage3_tmp_extract.joinpath("dev"))
-                unlink_try_hard(stage3_tmp_extract.joinpath("proc"))
-                unlink_try_hard(stage3_tmp_extract.joinpath("sys"))
+        with complete_step(f"Extracting {stage3_tar.name} to {cls.stage3_cache}"):
+            run([
+                "tar",
+                "--numeric-owner",
+                "-C", cls.stage3_cache,
+                "--extract",
+                "--file", stage3_tar_path,
+                "--exclude", "./dev",
+            ])
 
-                stage3_tmp_extract.joinpath(".cache_isclean").touch()
+            unlink_try_hard(cls.stage3_cache.joinpath("dev"))
+            unlink_try_hard(cls.stage3_cache.joinpath("proc"))
+            unlink_try_hard(cls.stage3_cache.joinpath("sys"))
```

Comment:
We're still doing this? I thought you commented elsewhere this was removed and replaced by excluding in `tar`?

Reply:
excluding `sys` breaks compiler! i suspect permission issue.